### PR TITLE
Fixes #75 Add Loose Mutable Mapping Merges

### LINKED ISSUE #75: Loose typechecking for MutableMapping objects in _check_typesafe
It would be nice to add the ability to check if obj1[key] and obj2[key] in _check_typesafe are functional equivalents, not just exact type equivalents, particularly for various types of dict like objects.

When attempting to merge two dicts, I encountered an issue where an object subclassing dict in src was unable to be merged into a dict object in dst.  I think a check to see if the two objects in question are instances of MutableMapping may be sufficient to get most cases.  It seems to me that two dict-like-objects should continue the recursive merge and not force a wholesale replacement of dst by src.

Perhaps an additional flag could be added to permit this loose functional equivalence matching or at the least MutableMapping instance matching.  As far as the resulting object type, I feel that merge should keep the dst's current object type, but perhaps that could be configured by a flag.

### PR DESCRIPTION
**Pull request by gotnone**
Changes to _check_typesafe and additional flag to permit merging of
MutableMapping objects at the same depth.

Issue #75 may encompass a larger set of functional equivalent object pairs, but this at least covers objects which are instances of MutableMapping.

Items that should be discussed:
1) Should the flags MERGE_TYPESAFE and MERGE_LOOSEDICT be mutually exclusive?  It seems like MERGE_TYPESAFE should imply strict type checking, in which case, MERGE_TYPESAFE should disable MERGE_LOOSEDICT and if both are selected, perhaps an error should be raised.

2) As far as the current implementation, I have no problem with putting the MERGE_LOOSEDICT check after the ...== MERGE_TYPESAFE code block, but prior to the ...!= MERGE_TYPESAFE block.  This prevents dst[v] from being popped and replaced by src[v], but allows MERGE_TYPESAFE to exhibit strict type checking, including in MutableMapping cases.

3) Does it make sense to eliminate the MERGE_LOOSEDICT flag and just include the ability to merge MutableMapping objects of differing types by default?  If so, we could set the condition to exit _check_typesafe as ...!= MERGE_TYPESAFE and ensure both obj1 and obj2 satisfy isinstance(..., MutableMapping)

### CODE REVIEW

**Comment by akesterson** [comment]
Looks good, thanks for the contribution. I agree that the two merge types should be mutually exclusive, and added logic to ensure that.

**Comment by akesterson** [comment]
While releasing 2.0 we found that LOOSEDICT and TYPESAFE weren't working the way we expected after this got merged. Specifically LOOSEDICT didn't seem necessary, but rather LOOSEDICT was enabling functionality that we actually expected to be the default behavior from the beginning. See https://github.com/akesterson/dpath-python/pull/111 for how this was modified.

TL;DR - LOOSEDICT is no longer a flag. Merging mutable sequences and mappings is now the default behavior. If you don't like this (most will like it), use TYPESAFE merge.
